src/AutoRest.CSharp/Common/AutoRest/Plugins/NewProjectScaffolding.cs

```diff
@@ -0,0 +1,429 @@
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http.Json;
+using System.Text;
+using System.Threading.Tasks;
+using AutoRest.CSharp.AutoRest.Communication;
+using AutoRest.CSharp.AutoRest.Plugins;
+using AutoRest.CSharp.Input;
+using AutoRest.CSharp.Output.Models.Types;
+using static System.Net.Mime.MediaTypeNames;
+
+namespace AutoRest.CSharp.Common.AutoRest.Plugins
+{
+    internal class NewProjectScaffolding
+    {
+        private string _serivceDirectoryName;
+        private string _projectDirectory;
+        private string _testDirectory;
+        private string _serviceDirectory;
+        private bool _isAzureSdk;
+
+        public NewProjectScaffolding()
+        {
+            _serivceDirectoryName = Path.GetFileName(Path.GetFullPath(Path.Combine(Configuration.AbsoluteProjectFolder, "..", "..")));
+            _projectDirectory = Path.Combine(Configuration.AbsoluteProjectFolder, "..");
+            _testDirectory = Path.Combine(Configuration.AbsoluteProjectFolder, "..", "tests");
+            _serviceDirectory = Path.Combine(Configuration.AbsoluteProjectFolder, "..", "..");
+            _isAzureSdk = Configuration.Namespace.StartsWith("Azure.");
+        }
+
+        public async Task<bool> Execute()
+        {
+            if (!_isAzureSdk)
+            {
+                //clean up old sln and csproj files
+                foreach (var file in Directory.GetFiles(Configuration.AbsoluteProjectFolder, "*.csproj"))
+                {
+                    File.Delete(file);
+                }
+                foreach (var file in Directory.GetFiles(_projectDirectory, "*.sln"))
+                {
+                    File.Delete(file);
+                }
+                foreach (var file in Directory.GetFiles(_testDirectory, "*.csproj"))
+                {
+                    File.Delete(file);
+                }
+            }
+
+            if (_isAzureSdk)
+                await WriteServiceDirectoryFiles();
+
+            await WriteSolutionFiles();
+
+            await WriteProjectFiles();
+
+            await WriteTestFiles();
+
+            return true;
+        }
+
+        private async Task WriteServiceDirectoryFiles()
+        {
+            //TODO handle existing ci where multiple projects are in the same service directory
+            await File.WriteAllBytesAsync(Path.Combine(_serviceDirectory, "ci.yml"), Encoding.ASCII.GetBytes(GetCiYml()));
+        }
+
+        private async Task WriteTestFiles()
+        {
+            if (_isAzureSdk)
+            {
+                Directory.CreateDirectory(Path.Combine(_testDirectory, "SessionRecords"));
+            }
+            await File.WriteAllBytesAsync(Path.Combine(_testDirectory, $"{Configuration.Namespace}.Tests.csproj"), Encoding.ASCII.GetBytes(GetTestCsproj()));
+            //WriteTestBaseClass(autoRest);
```

Comment:
nit:  Should we remove these before merging?

Reply:
I will mark them as TODO so its clear this is future work